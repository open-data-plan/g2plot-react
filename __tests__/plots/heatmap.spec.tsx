import { render } from '@testing-library/react';
import axios from 'axios';
import React from 'react';
import HeatmapChart from '../../src/plots/heatmap';

describe('HeatmapChart', () => {
  test('should render without crashed', async () => {
    const res = await axios.get(
      'https://gw.alipayobjects.com/os/basement_prod/a719cd4e-bd40-4878-a4b4-df8a6b531dfe.json',
    );
    const result = render(
      <HeatmapChart
        {...{
          width: 650,
          height: 500,
          autoFit: false,
          data: res.data,
          xField: 'Month of Year',
          yField: 'District',
          colorField: 'AQHI',
          color: ['#174c83', '#7eb6d4', '#efefeb', '#efa759', '#9b4d16'],
          meta: {
            'Month of Year': {
              type: 'cat',
            },
          },
        }}
      />,
    );

    expect(result.container).toMatchSnapshot();
  });
});
