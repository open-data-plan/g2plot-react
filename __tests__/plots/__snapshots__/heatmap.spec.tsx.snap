// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeatmapChart should render without crashed 1`] = `
<div>
  <div
    data-chart-source-type="G2Plot"
  >
    <div
      style="position:relative;"
    >
      <canvas
        height="500"
        style="width: 650px; height: 500px; display: inline-block; vertical-align: middle;"
        width="650"
      />
    </div>
  </div>
</div>
`;
